src/app/Main/Body/QueryPage/Services/TableServices/calculateRowColSums.service.ts

```diff
@@ -33,58 +33,53 @@ function calcSums(
     });
 
     const rowWithColSums: Record<string, string> = {};
-    Object.keys(colSums).forEach((k) => {
-        rowWithColSums.name = "colSum";
-        rowWithColSums[k] = mapColSums(colSums, k);
+    rowWithColSums.name = "colSum";

```

Comment:
I added a comment about this and removed it because this might cause a problem in the container getting the sumRow from the table:
If the table is empty, the sumRow is set to a defined object now. So getting the sumRow in the container: rows.filter(name==="colSum")[0] would not return undefined and the sumRowViewComponent could show something stupid, i guess.
So please check what happens if the table is empty and if any exceptions are thrown in the console.

Reply:
So i didn't checked the empty table logic. Probably the container should just omit the rowWithColSums component, so it never get's an undefined rowWithColSums input. Also possible that the code path with the undefined rowWithColSums input is never reached.

Reply:
There are no problems with an empty table, because then a text and not the table is displayed. There are also no warnings in the console.

Reply:
👍